File: app/src/main/java/com/samhalperin/cooperhewitt/application/BaseActivity.java
package com.samhalperin.cooperhewitt.application;

import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import com.samhalperin.cooperhewitt.R;

public class BaseActivity extends AppCompatActivity {

    protected void setup(int contentView) {
        setContentView(contentView);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        getSupportActionBar().setDisplayUseLogoEnabled(true);
        getSupportActionBar().setLogo(R.mipmap.ic_launcher);
        getSupportActionBar().setDisplayShowTitleEnabled(false);
    }
}

File: app/src/main/java/com/samhalperin/cooperhewitt/application/BaseActivity.kt
package com.samhalperin.cooperhewitt.application

import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.samhalperin.cooperhewitt.R
import kotlinx.android.synthetic.main.activity_main.*

open class BaseActivity : AppCompatActivity() {

    protected fun setup(contentView: Int) {
        setContentView(contentView)
        setSupportActionBar(toolbar as Toolbar)
        supportActionBar?.setHomeButtonEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)
        supportActionBar?.setDisplayUseLogoEnabled(true)
        supportActionBar?.setLogo(R.mipmap.ic_launcher)
        supportActionBar?.setDisplayShowTitleEnabled(false)
    }
}
